File: flyweight/java/Box.java
package Flyweight;

// Nesnenin ortak özelliklerini tutan sınıftır.
// UML diyagramındaki Flyweight yapısına denk gelmektedir.
abstract class Box
{
    // Intrinsic değerler.
    public int width;
    public int height;
    public BoxColor boxColor;

    public abstract void draw(int locationX, int locationY);
}

File: flyweight/kotlin/Box.kt
package Flyweight

// Nesnenin ortak özelliklerini tutan sınıftır.
// UML diyagramındaki Flyweight yapısına denk gelmektedir.
internal abstract class Box {
    // Intrinsic değerler.
    var width: Int = 0
    var height: Int = 0
    var boxColor: BoxColor? = null

    abstract fun draw(locationX: Int, locationY: Int)
}
